prusti-interface/src/specs/external.rs

```diff
@@ -0,0 +1,98 @@
+use rustc_hir::intravisit;
+use rustc_hir::def_id::DefId;
+use rustc_middle::hir::map::Map;
+use rustc_middle::ty::TyCtxt;
+use rustc_span::Span;
+
+use std::collections::HashMap;
+use crate::specs::typed::ExternSpecificationMap;
+
+pub struct ExternSpecResolver<'tcx> {
+    tcx: TyCtxt<'tcx>,
+    /// Map of def id of the real function to a tuple containing the def id of the implementing type (or none)
+    /// mapped to the fake function with the specifications. Mapped to a tuple to account for trait implementations.
+    extern_fn_map: ExternSpecificationMap<'tcx>,
```

Comment:
The comment looks very confusing to me (is it describing three nested maps?). I think it should be rewritten.

Reply:
How about this:

```
    /// Maps real functions (keyed by their `def_id`) to Prusti-generated fake
    /// functions with specifications. The mapping may also optionally contain
    /// the `def_id` of the implementing type to account for trait
    /// implementations.
```

Reply:
Much better :+1: